pallets/node-authorization/src/lib.rs

```diff
@@ -150,6 +150,7 @@ pub mod pallet {
 	}
 
 	#[pallet::error]
+	#[derive(PartialEq)]
```

Comment:
Is this required?

Reply:
yes @vatsa287 because `generate_peer_id` is returning `Result<PeerId, Error<T>>` and to compare return value of `generate_peer_id`, error must have `#[derive(PartialEq)`
I tried different ways but none worked, I am open to suggestions to change it. I am a newbie in rust so could really use suggestions. Thanks.

Reply:
Ack. It is required for a non-extrinsic function based error comparisons.